app/components/Form/ProjectManagerForm.tsx

```diff
@@ -38,27 +69,168 @@ const ProjecManagerForm: React.ForwardRefRenderFunction<any, Props> = (
     };
-    return initialSchema as JSONSchema7;
+
+    return schema as JSONSchema7;
   }, [allCifUsers]);
 
-  return <FormBase {...props} ref={ref} schema={schema} uiSchema={uiSchema} />;
+  // Add a manager to the project revision
+  const [applyAddManagerToRevision] = useAddManagerToRevisionMutation();
+  const addManager = (data: any) => {
+    applyAddManagerToRevision({
+      variables: {
+        projectRevision: projectRevision.id,
+        projectRevisionId: projectRevision.rowId,
+        newFormData: data,
+      },
+    });
+  };
+
+  // Delete a manager from the project revision
+  const [discardFormChange] = useDeleteManagerFromRevisionMutation();
+  const deleteManager = (id: string) => {
+    discardFormChange({
+      variables: {
+        input: {
+          id: id,
+        },
+        projectRevision: projectRevision.id,
+      },
+    });
+  };
+
+  const [applyUpdateFormChangeMutation] = useDebouncedMutation(
+    updateFormChangeMutation
+  );
+  // Update an existing project_manager form change if it exists, otherwise create one
+  const createOrUpdateFormChange = (
+    formChangeId: string,
+    labelId: number,
+    change: any
+  ) => {
+    const data = {
+      ...change,
+      projectManagerLabelId: labelId,
+      projectId: projectRevision.projectFormChange.formDataRecordId,
+    };
+
+    // If a form_change already exists, and the payload contains a cifUserId update it
+    if (formChangeId && change.cifUserId) {
+      applyUpdateFormChangeMutation({
+        variables: {
+          input: {
+            id: formChangeId,
+            formChangePatch: {
+              newFormData: data,
+            },
+          },
+        },
+        optimisticResponse: {
+          updateFormChange: {
+            formChange: {
+              id: formChangeId,
+              newFormData: data,
+            },
+          },
+        },
+        debounceKey: formChangeId,
+      });
+      // If a form_change does not exist, and the payload contains a cifUserId create a form_change record
+    } else if (change.cifUserId) {
+      addManager(data);
+    }
+    // If a form_change exists, and the payload does not contain a cifUserId delete it
+    else if (formChangeId && !change.cifUserId) {
+      deleteManager(formChangeId);
+    }
+  };
+
+  props.setValidatingForm({
+    selfValidate: () => {
+      return Object.keys(formRefs.current).reduce((agg, formId) => {
+        const formObject = formRefs.current[formId];
+        return [...agg, ...validateFormWithErrors(formObject)];
+      }, []);
+    },
+  });
+
+  return (
+    <>
+      <Grid cols={10} align="center">
+        <Grid.Row>
+          <Grid.Col span={10}>
+            <FormBorder title="Project Managers">
+              {projectRevision.managerFormChanges.edges.map(({ node }) => (
+                <React.Fragment key={node.projectManagerLabel.id}>
+                  <Grid.Row>
+                    <label>{node.projectManagerLabel.label}</label>
```

Comment:
```suggestion
                    <label htmlFor={your_select_id}>{node.projectManagerLabel.label} (optional)</label>
```
In the current behaviour, all managers are optional, so this should be reflected in the label. 
We do have a `FieldLabel` component, which could be reused here. At least the `getRequiredLabel` function could be reused.